azuki-script-generation/src/main/kotlin/com/anaplan/engineering/azuki/script/generation/ScriptGenerationCheck.kt

```diff
@@ -2,32 +2,44 @@ package com.anaplan.engineering.azuki.script.generation
 
 import com.anaplan.engineering.azuki.core.system.Check
 
-sealed interface ScriptGenerationCheck : Check
+interface ScriptGenerationCheck<E : ScriptGenerationEnvironment> : Check {
 
-/**
- * A basic, non-composable script generation check that does not need an environment.
- */
-interface BasicScriptGenerationCheck : ScriptGenerationCheck {
-
-    fun getCheckScript(): String
+    /**
+     * Gets the script for this check without trying to compose it.
+     * May fail if the check is impossible to express as DSL without composition.
+     */
+    fun getCheckScript(environment: E): String
 }
 
-/**
- * A basic, non-composable script generation check that uses an environment.
- */
-interface EnvScriptGenerationCheck<in E : ScriptGenerationEnvironment> : ScriptGenerationCheck {
+interface ComposableScriptGenerationCheck<E : ScriptGenerationEnvironment> : ScriptGenerationCheck<E> {
 
-    fun getCheckScript(environment: E): String
+    /**
+     * Composes the check into the environment.
```

Comment:
⛏️ I find this a bit confusing... what does it mean to compose a check into an environment? I understand this to mean that the environment itself is in a sense a check, or at least that it performs a much more central role than what I would expect (i.e. to just hold state).

Reply:
I realise this wording was in the previous version as well... it slipped from my previous review.

Reply:
The environment has basically subsumed the old 'checkstate' idea from current Azuki, so the idea is that you're submitting the check into the environment's internal tracking of composable checks so that it can possibly be reconciled with other similar checks and later pop out as a list of zero, one, or more merged final checks.

Reply:
It's now `registerComposable`.  (I'm not 100% wedded to this name either... `registerForComposition`, `addForComposition`, etc... might work well too.  The key idea is that something is being *stored* in the environment for the purpose of *eventually* producing a composed check.)